File: examples/Training/python/Connect_To_OMERO.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

#
# Copyright (C) 2014 University of Dundee & Open Microscopy Environment.
#                    All Rights Reserved.
# Use is subject to license terms supplied in LICENSE.txt
#

"""
FOR TRAINING PURPOSES ONLY!
"""


from Parse_OMERO_Properties import USERNAME, PASSWORD, HOST, PORT
from omero.gateway import BlitzGateway

if __name__ == '__main__':
    """
    NB: This block is only run when calling this file directly
    and not when imported.
    """

    # Connect to the Python Blitz Gateway
    # =============================================================
    # Make a simple connection to OMERO, printing details of the
    # connection. See OmeroPy/Gateway for more info
    conn = BlitzGateway(USERNAME, PASSWORD, host=HOST, port=PORT)
    connected = conn.connect()

    # Check if you are connected.
    # =============================================================
    if not connected:
        import sys
        sys.stderr.write(
            "Error: Connection not available, please check your user name and"
            " password.\n")
        sys.exit(1)

    # Using secure connection.
    # =============================================================
    # By default, once we have logged in, data transfer is not encrypted
    # (faster)
    # To use a secure connection, call setSecure(True):

    # conn.setSecure(True)         # <--------- Uncomment this

    # Current session details
    # =============================================================
    # By default, you will have logged into your 'current' group in OMERO.
    # This can be changed by switching group in the OMERO.insight or OMERO.web
    # clients.

    user = conn.getUser()
    print "Current user:"
    print "   ID:", user.getId()
    print "   Username:", user.getName()
    print "   Full Name:", user.getFullName()

    print "Member of:"
    for g in conn.getGroupsMemberOf():
        print "   ID:", g.getName(), " Name:", g.getId()
    group = conn.getGroupFromContext()
    print "Current group: ", group.getName()

    # List the group owners and other members
    owners, members = group.groupSummary()
    print "   Group owners:"
    for o in owners:
        print "     ID: %s %s Name: %s" % (
            o.getId(), o.getOmeName(), o.getFullName())
    print "   Group members:"
    for m in members:
        print "     ID: %s %s Name: %s" % (
            m.getId(), m.getOmeName(), m.getFullName())

    print "Owner of:"
    for g in conn.listOwnedGroups():
        print "   ID: ", g.getName(), " Name:", g.getId()

    # New in OMERO 5
    print "Admins:"
    for exp in conn.getAdministrators():
        print "   ID: %s %s Name: %s" % (
            exp.getId(), exp.getOmeName(), exp.getFullName())

    # The 'context' of our current session
    ctx = conn.getEventContext()
    # print ctx     # for more info

    # Close connection:
    # =================================================================
    # When you are done, close the session to free up server resources.
    conn._closeSession()

File: examples/Training/matlab/ConnectToOMERO.m
% Copyright (C) 2011-2015 University of Dundee & Open Microscopy Environment.
% All rights reserved.
%
% This program is free software; you can redistribute it and/or modify
% it under the terms of the GNU General Public License as published by
% the Free Software Foundation; either version 2 of the License, or
% (at your option) any later version.
%
% This program is distributed in the hope that it will be useful,
% but WITHOUT ANY WARRANTY; without even the implied warranty of
% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
% GNU General Public License for more details.
%
% You should have received a copy of the GNU General Public License along
% with this program; if not, write to the Free Software Foundation, Inc.,
% 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.


try
    %% Connect to a server
    % Use the ice.config file defined in the path
    client = loadOmero();
    p = parseOmeroProperties(client);
    fprintf(1, 'Created connection to %s\n', p.hostname);
    
    % Alternate ways to create clients
    % client = loadOmero(hostname);
    % client = loadOmero('path/to/ice.config');
    
    % Information to edit
    username = p.username;
    password = p.password;
   
    % Create OMERO session
    session = client.createSession(username, password);
    adminService = session.getAdminService();
    
    % Retrieve the information of the user
    userId = adminService.getEventContext().userId;
    userName = char(adminService.getEventContext().userName);
    
    % The group the user is currently logged in i.e. his/her default group
    groupId = adminService.getEventContext().groupId;
    groupName = char(adminService.getEventContext().groupName);
    
    fprintf(1, 'Created session for user %s (id: %g) using group %s (id: %g)\n',...
        userName, userId, groupName, groupId);
    
    %necessary to keep the proxy alive. part of the omero-package
    disp('Keep session alive');
    t = omeroKeepAlive(client);
    stop(t);
    delete(t);
    
    %% Admin service
    % Retrieve  the identifier of the groups the user is member/owner of
    user = adminService.getExperimenter(userId);
    groupIds1 = toMatlabList(adminService.getMemberOfGroupIds(user));
    groupIds2 = toMatlabList(adminService.getLeaderOfGroupIds(user));
    
    % List all groups the user is member of
    disp('Group membership');
    for groupId = groupIds1'
        group = adminService.getGroup(groupId);
        fprintf(1, ' Group %s (id: %g, type: %s)\n',...
            char(group.getName().getValue()), groupId,...
            char(group.getDetails().getPermissions()));
    end

    % List all groups the user is owner of
    disp('Group ownership');
    for groupId = groupIds2'
        group = adminService.getGroup(groupId);
        fprintf(1, ' Group %s (id: %g, type: %s)\n',...
            char(group.getName().getValue()), groupId,...
            char(group.getDetails().getPermissions()));
    end
    
    %% Unencrypted session
    % Create an unsecure client and session
    % Use this session to speed up data transfer since there will be no
    % encryption
    unsecureClient = client.createClient(false);
    sessionUnencrypted = unsecureClient.getSession();
    fprintf(1, 'Created encryted session for user %s (id: %g)\n',...
        userName, userId);
    
catch err
    client.closeSession();
    unsecureClient.closeSession();
    throw(err);
end

% Close the sessions
client.closeSession();
unsecureClient.closeSession();
